annet/gen.py

```diff
@@ -232,11 +233,30 @@ def _old_new_per_device(ctx: OldNewDeviceContext, device: Device, filterer: Filt
 
         entire_results = res.entire_results
         json_fragment_results = res.json_fragment_results
+        old_json_fragment_files = old_files.json_fragment_files
+
         new_files = res.new_files()
         new_json_fragment_files = res.new_json_fragment_files(old_files.json_fragment_files)
+
         if ctx.args.acl_safe:
             safe_new_files = res.new_files(safe=True)
 
+        if not ctx.args.no_acl:
+            filters = build_filter_text(filterer, device, ctx.stdin, ctx.args, ctx.config).split('\n')
+
+            for file_name in new_json_fragment_files:
+                new_json_fragment_files = _update_json_config(
+                    new_json_fragment_files,
+                    file_name,
+                    jsontools.apply_acl_filters(new_json_fragment_files[file_name][0], filters)
```

Comment:
Never empty?

Reply:
As far as i can see, yes. First item of tuple always exists ([creating here](https://github.com/annetutil/annet/blob/eca66a47b83e9b3aab431f9d086fbe21bd5a7b0d/annet/generators/__init__.py#L180)). At worst, it can be empty dict, but it will be.